### DISCUSSION TITLE
Mojo does not allow dynamic value in default argument

### DISCUSSION THREAD

**Discussion by ivellapillil** [modular/modular#801] [General]
The following code works in python, but not in mojo:

```
def main():
    
    for i in range(5):
        def afn(x, i = i):
            print(i)
        
```

I get:

```
error: cannot use a dynamic value in default argument
        def afn(x, i = i):
```

Is this "incompatibility" going to stay? IIRC, it is somewhat common idiom in Python to overcome lazy evaluation...

**Comment by rarebreed** [comment]
This limitation is described in the [rough edges section](https://docs.modular.com/mojo/roadmap.html#small-independent-features) of the documentation (first bullet point in the small independent features).

It is in Modular's plans to make Mojo [eventually compatible with python](https://docs.modular.com/mojo/why-mojo.html#compatibility-with-python).  But mojo is very young, so I think it will take quite some time to get there.  I hope Modular will open source it once they have a good solid foundation to build on, and the community can help mojo become a superset of python faster.

**Comment by lattner** [comment]
Closing this just because it is tracked in rough edges already, we definitely need to support this in time
